test/demo/fake_server_test.exs

```diff
@@ -0,0 +1,37 @@
+defmodule Wanda.Executions.FakeServerTest do
+  use Wanda.DataCase
+
+  import Mox
+  import Wanda.Factory
+
+  alias Trento.Checks.V1.{
+    ExecutionCompleted,
+    ExecutionStarted
+  }
+
+  alias Wanda.Executions.{
+    Execution,
+    FakeServer
+  }
+
+  setup :verify_on_exit!
+
+  test "start_execution publishes execution started and completed messages" do
```

Comment:
I think we should mock the `sleep` function, otherwise this test will take 2 seconds everytime, which is already more than the whole suite